src/pharmpy/plugins/nlmixr/model.py

```diff
@@ -4,52 +4,60 @@
 import uuid
 import warnings
 from dataclasses import dataclass, replace
-from pathlib import Path
-from typing import Optional
+from pathlib import Path, PosixPath
+from typing import Optional, Union
 
 import pharmpy.model
 from pharmpy.deps import pandas as pd
-from pharmpy.deps import sympy, sympy_printing
-from pharmpy.model import Assignment
+from pharmpy.deps import sympy_printing
 from pharmpy.modeling import (
+    append_estimation_step_options,
+    drop_columns,
     get_evid,
     get_sigmas,
     get_thetas,
     set_evaluation_step,
+    translate_nmtran_time,
     update_inits,
     write_csv,
 )
 from pharmpy.results import ModelfitResults
+from pharmpy.tools import fit
 
+from .CodeGenerator import CodeGenerator
+from .create_ini import add_eta, add_sigma, add_theta
+from .create_model_block import add_ode, add_statements
+from .name_mangle import name_mangle
+from .sanity_checks import check_model, print_warning
 
-class CodeGenerator:
-    def __init__(self):
-        self.indent_level = 0
-        self.lines = []
 
-    def indent(self):
-        self.indent_level += 4
+def convert_model(
+    model: pharmpy.model.Model, keep_etas: bool = False, skip_check: bool = False
+) -> pharmpy.model.Model:
+    """
+    Convert a NONMEM model into an nlmixr model
 
-    def dedent(self):
-        self.indent_level -= 4
+    Parameters
+    ----------
+    model : pharmpy.model.Model
+        A NONMEM pharmpy model object
+    keep_etas : bool, optional
+        Decide if NONMEM estimated thetas are to be used. The default is False.
+    skip_check : bool, optional
+        Skip determination of error model type. Could speed up conversion. The default is False.
 
-    def add(self, line):
-        self.lines.append(f'{" " * self.indent_level}{line}')
+    Returns
+    -------
+    pharmpy.model
```

Comment:
Should it be pharmpy.model.Model?

Reply:
Yes, my mistake, this has been corrected to `pharmpy.model.Model`